### PR TITLE
DEVX-6810: Add support for E2EE

### PR DESCRIPTION
**Pull request by superchilled**
## Description

This PR does the following:

- Updates the `Opentok#create_session` method to:
  - Add an option for `:e2ee` to `valid_opts`
  - Adds conditional checks to the method ensure `:e2ee` is not set to `true` if `:media_mode` is `:relayed` or if `:archive_mode` is `:always`
  - Update the docs comments to provide information on the new param and a link to the relevant OpenTok documentation
 
- Adds unit tests for the new functionality described above

## Motivation and Context

It implements a feature in the SDK which is being added to the API.

## How Has This Been Tested?

Added a number tests for calling `Opentok#create_session` with the new `:e2ee` param set, as well as tests for invalid configuration options

## Example Output or Screenshots (if appropriate):

n/a

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: lib/opentok/opentok.rb [hunk 1]
```diff
@@ -148,7 +148,7 @@ def initialize(api_key, api_secret, opts={})
     def create_session(opts={})
 
       # normalize opts so all keys are symbols and only include valid_opts
-      valid_opts = [ :media_mode, :location, :archive_mode ]
+      valid_opts = [ :media_mode, :location, :archive_mode, :e2ee ]
```
Comment (jeffswartz) [inline lib/opentok/opentok.rb:151]: Please add docs to the comment above this (on line 146):

```
    # @option opts [Symbol] : e2ee Determines whether to enable
    # {https://tokbox.com/developer/guides/end-to-end-encryption end-to-end encryption}
    # for the session.

````

**Review by jeffswartz — CHANGES_REQUESTED** [review]
Should we rev the version to 4.6.0 in the version.rb file? And note changes in the CHANGES.md file?

File: lib/opentok/opentok.rb (hunk 1 above)
Comment (superchilled) [reply lib/opentok/opentok.rb:151]: Thanks @jeffswartz. This is still a draft PR at this stage. I'll add the docs comments and also update the README before finalizing. I'm also planning to add some validation on the opts when creating a session, but am awaiting further info on this (my assumption is that `e2ee` should **not** be set to `true` if `archive_mode` is `always`, or if `media_mode` is relayed).

Will rev version and update CHANGES if this is going out as a stand-alone release. If it gets batched together in a release with the Audio Connector feature, I'll do that separately to cover both features.

File: lib/opentok/opentok.rb [hunk 2]
```diff
@@ -144,11 +144,16 @@ def initialize(api_key, api_secret, opts={})
     #     automatically (<code>:always</code>) or not (<code>:manual</code>). When using automatic
     #     archiving, the session must use the <code>:routed</code> media mode.
     #
+    # @option opts [true, false] :e2ee
+    #     (Boolean, optional) — whether the session is end-to-end encrypted from client to client (default: false).
+    #     Should not be set to `true` if `:media_mode` is `:relayed` or if `:archive_mode` is `:always`.
```
Comment (jeffswartz) [inline lib/opentok/opentok.rb:149]:
```suggestion
    #     (Boolean, optional) — Whether the session uses end-to-end encryption from client to client (default: false).
    #     This should not be set to `true` if `:media_mode` is `:relayed` or if `:archive_mode` is `:always`.
```

**Pull request merged**